tests/com/carolinarollergirls/scoreboard/jetty/WSTest.java

```diff
@@ -0,0 +1,68 @@
+package com.carolinarollergirls.scoreboard.jetty;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class WSTest {
+    private WS.PathTrie pt;
+
+    @Before
+    public void setup() {
+        pt = new WS.PathTrie();
+    }
+
+    @Test
+    public void path_trie_simple() {
+        pt.add("ScoreBoard.Period");
+
+        assertTrue(pt.covers("ScoreBoard.Period"));
+        assertTrue(pt.covers("ScoreBoard.Period(1)"));
+        assertTrue(pt.covers("ScoreBoard.Period.Jam"));
+        assertTrue(pt.covers("ScoreBoard.PeriodFoo"));
+
+        assertFalse(pt.covers("ScoreBoard.Perioc"));
+        assertFalse(pt.covers("ScoreBoard.Perioe"));
+        assertFalse(pt.covers("ScoreBoard(1).Perioe"));
+        assertFalse(pt.covers("ScoreBoard.Perio"));
+        assertFalse(pt.covers("Scoreboard.Period"));
+    }
+
+    public void path_trie_overlap() {
+        pt.add("ScoreBoard.Period(1).Jam(1).StarPass");
+        pt.add("ScoreBoard.Period");
+
+        assertTrue(pt.covers("ScoreBoard.Period"));
+        assertTrue(pt.covers("ScoreBoard.Period(1)"));
+        assertTrue(pt.covers("ScoreBoard.Period(1).Jam(1).StarPass"));
+    }
+
+    public void path_trie_id_glob() {
+        pt.add("ScoreBoard.Period(*).Jam(1).Foo(*).Bar");
+
+        assertTrue(pt.covers("ScoreBoard.Period(1)"));
+        assertTrue(pt.covers("ScoreBoard.Period(1).Jam(1).Foo(2).Bar"));
+        assertTrue(pt.covers("ScoreBoard.Period(1).Jam(1).Foo(2).Bar.Baz"));
+        assertTrue(pt.covers("ScoreBoard.Period(1).Jam(1).Foo(2).Bar(zzz)"));
+
+        assertFalse(pt.covers("ScoreBoard.Period"));
+        assertFalse(pt.covers("ScoreBoard.Period("));
+        assertFalse(pt.covers("ScoreBoard.Period(1).Jam(2).Foo(2).Bar"));
+        assertFalse(pt.covers("ScoreBoard.Period(1).Jam(1).Foo(2)"));
+        assertFalse(pt.covers("ScoreBoard.Period(1).Jam(1).TeamJam(1).Foo(2)"));
+    }
+
+    public void path_trie_non_id_glob() {
+        pt.add("ScoreBoard.Period*");
+
+        assertTrue(pt.covers("ScoreBoard.Period*"));
+        assertTrue(pt.covers("ScoreBoard.Period*a"));
+
+        assertFalse(pt.covers("ScoreBoard.Period"));
+        assertFalse(pt.covers("ScoreBoard.Period("));
+        assertFalse(pt.covers("ScoreBoard.Period(1).Jam(2).Foo(2).Bar"));
+    }
+
+}
```

Comment:
Maybe add a test where you combine `Period(1).Foo` and `Period(*).Bar`.

Reply:
Did you miss this one?

Reply:
path_trie_overlap_glob does this now.